src/parsec_client/operations/attest_key_params.md

```diff
@@ -0,0 +1,64 @@
+# Key Attestation Parameters
+
+This page defines the mechanism-specific inputs and outputs of `AttestKey`. For an in-depth look at
+the mechanisms and hardware tokens that we've considered, you can read our write-up
+[here](https://drive.google.com/file/d/11jLLv8zN_xRunj9BEkk_L_tj5DiQOtkG/view?usp=sharing).
+
+Each mechanism comes with its own definitions for `AttestationMechanismParams` and
```

Comment:
I want to make sure I understand what is meant by "each mechanism comes with its own definitions". Do we really mean definitions? Or just interpretations of the fields? There can only be one definition of the structure in terms of the Protobuf messages. Will the two fields `credential_blob` and `secret` be common, but with different interpretations as to their meaning? Or are these two fields inherently specific to the TPM mechanism only? If it's the latter, then how do we expect Protobuf definitions to look when multiple mechanism are supported?

Reply:
I do mean that each mechanisms essentially defines a completely new set of fields. You can see how I'm thinking of implementing the protobuf contracts in section 3 of [the investigation doc](https://docs.google.com/document/d/11jLLv8zN_xRunj9BEkk_L_tj5DiQOtkG/edit?usp=sharing&ouid=110488261398236555557&rtpof=true&sd=true). We'd use the `oneof` protobuf feature that allows you to define a field essentially as an "enum" of different message types. So for `ActivateCredential` you get a "nested" message with those two fields (`credential_blob` and `secret`), for others you'd get something else. There are multiple advantages to doing it this way, including this flexibility/liberty of adding new definitions easily and in a backwards compatible manner.